### PR TITLE
Implement IDF-1 global weighting (fix #209)

### LINKED ISSUE #209: Recommender: Inverted_index using idf does not store the first ID.
* config:

  ```
  {
    "converter" : {
      "string_rules":[
        {"key" : "*", "type" : "str", "sample_weight":"bin", "global_weight" : "idf"}
      ]
    },
    "method": "inverted_index"
  }
  ```

The steps are the following:

1. update_row("ID", {"key": "id"})
2. get_all_rows
    * expected: ["ID"]
    * actual: []

### PR DESCRIPTION
**Pull request by kmaehashi**
Fix #209

### CODE REVIEW

**Comment by kmaehashi** [comment]
# Precision

The precision is slightly worse than existing TF-IDF. (note: Results other than TF-IDF1 are quoted from #295)

Experimented using Jubatus Tutorial Python with Perceptron:

* **TF-IDF1: 66.1% (OK = 4977 / NG = 2555)**
* Bin-Bin: 63.5% (OK = 4780 / NG = 2752)
* TF-IDF: 68.1% (OK = 5129 / NG = 2403)
* Bin-BM25: 72.3% (OK = 5446 / NG = 2086)
* TF-BM25: 69.4% (OK = 5230 / NG = 2302)

Experimented using Jubatus Tutorial Python with AROW (`regularization_weight == 0.1`):

* **TF-IDF1: 80.6% (OK = 6072 / NG = 1460)**
* Bin-Bin: 75.6% (OK = 5691 / NG = 1841)
* TF-IDF: 81.9% (OK = 6170 / NG = 1362)
* Bin-BM25: 83.6% (OK = 6296 / NG = 1236)
* TF-BM25: 83.7% (OK = 6304 / NG = 1228)

File: jubatus/core/fv_converter/weight_manager.cpp [hunk 1]
```diff
@@ -122,10 +125,11 @@ void weight_manager::get_weight(common::sfv_t& fv) const {
 
 double weight_manager::get_global_weight_idf(
     const std::string& key,
+    double inflate_idf,
     double sample_weight) const {
   double doc_count = get_document_count();
   double doc_freq = get_document_frequency(key);
-  return std::log((doc_count + 1) / (doc_freq + 1)) * sample_weight;
+  return (std::log((doc_count + 1) / (doc_freq + 1)) + inflate_idf) * sample_weight;
```
Comment (TkrUdagawa) [inline jubatus/core/fv_converter/weight_manager.cpp]: cpplint error (Lines should be <= 80 characters long)

File: jubatus/core/fv_converter/weight_manager_test.cpp [hunk 2]
```diff
@@ -74,6 +75,9 @@ TEST(weight_manager, trivial) {
         ((/* tf = */ 4.0 * (1.2 + 1)) /
         (/* tf = */ 4.0 + 1.2 * (1 - 0.75 + 0.75 *
         (9.0 / 9.0)))), fv[5].second);
+
+    // String features weighted by bin-idf1
+    EXPECT_FLOAT_EQ(1.0 * (std::log((3.0 + 1) / (1.0 + 1)) + 1.0), fv[6].second);
```
Comment (TkrUdagawa) [inline jubatus/core/fv_converter/weight_manager_test.cpp]: cpplint error (Lines should be <= 80 characters long)

**Comment by TkrUdagawa** [comment]
LGTM except above lint errors.

**Comment by kmaehashi** [comment]
@TkrUdagawa Thanks, fixed.

**Comment by TkrUdagawa** [comment]
:+1:

**Pull request merged**